scripts/artifacts-building/containers/artifact-build-chroot.yml

```diff
@@ -0,0 +1,160 @@
+---
+# Copyright 2016, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- name: Create artifact container
+  hosts: localhost
+  connection: local
+  tasks:
+    #- name: Remove cache
+    #  file:
+    #    path: "/var/cache/lxc/download/ubuntu/trusty/amd64/default/"
+    #    state: absent
+    - name: Remove base container
+      lxc_container:
+        name: "LXC_NAME"
+        state: absent
+    - name: Create base container
+      lxc_container:
+        name: "LXC_NAME"
+        template: "download"
+        state: "stopped"
+        backing_store: "dir"
+        template_options: >
+          --dist {{ ansible_distribution | lower }}
+          --release {{ ansible_distribution_release | lower }}
+          --arch "{{ architecture_mapping.get( ansible_architecture ) }}"
+          --force-cache
+          --server images.linuxcontainers.org
+          --keyserver hkp://p80.pool.sks-keyservers.net:80
+    - name: Prepare group of master lxc hosts
+      add_host:
+        name: "/var/lib/lxc/LXC_NAME/rootfs/"
+        groups: "lxc_container_artifact_chroots"
+        container_name: "LXC_NAME"
+        physical_host: "localhost"
+  vars:
+    architecture_mapping:
+      x86_64: amd64
+      ppc64le: ppc64el
+  tags:
+    - always
+
+- name: Run role
+  hosts: lxc_container_artifact_chroots
+  gather_facts: True
+  connection: chroot
+  pre_tasks:
+    - name: Set facts to pass some variables that wouldn't work #like '{{ image_name }}_developer_mode'
+      set_fact: '{{ image_name }}_{{ item.key }}="{{ item.value }}"'
+      with_dict: "{{ role_vars }}"
+      tags:
+        - always
+    - name: Configure chroot
+      shell: |
+        if [ -a /etc/resolv.conf ]; then
+          mv /etc/resolv.conf /etc/resolv.conf.org
+        fi
+        echo "nameserver 8.8.8.8" > /etc/resolv.conf
+        mount -t proc proc /proc || true
+      tags:
+        - always
+  roles:
+    - role: "{{ role_name }}"
+  post_tasks:
+    - name: chroot resolv cleanup
+      shell: | 
+        if [ -a /etc/resolv.conf.org ]; then
+          mv /etc/resolv.conf.org /etc/resolv.conf
+        else
+          rm -f /etc/resolv.conf
+        fi
+        umount /proc || true        
```

Comment:
We should probably also do 'apt-clean' afterwards to clear the cache. Maybe also ditch the pip cache?